pkg/cmd/pulumi/new.go

```diff
@@ -457,6 +470,53 @@ func runNew(ctx context.Context, args newArgs) error {
 	return nil
 }
 
+// Retrieve a Private template from the given Pulumi Cloud URL **including an auth token for Pulumi Cloud**.
+//
+// If we cannot find current credentials for the given Pulumi Cloud URL we ask the user to login to that
+// URL using `pulumi login <cloud-url> --set-current=false`. We do this an extra safey measure to ensure
+// the user is aware of host they are trying to retrieve the template from and will be sending some kind
+// of crendetials to.
+func retrievePrivatePulumiCloudTemplate(templateURL string) (workspace.TemplateRepository, error) {
+	u, err := url.Parse(templateURL)
+	if err != nil {
+		return workspace.TemplateRepository{}, fmt.Errorf("parsing template URL: %w", err)
+	}
+	// Docs convention is to store the cloud URL with the protocol.
+	// e.g. `pulumi login https://api.pulumi.com` or pulumi login `https://api.acme.org`
+	templatePulumiCloudHost := "https://" + u.Host
+
+	howToLoginMessage := fmt.Sprintf(`Please login to the %s pulumi cloud backend to retrieve this template:
+
+pulumi login %s --set-current=false
+	`, templatePulumiCloudHost, templatePulumiCloudHost)
+
+	account, err := workspace.GetAccount(templatePulumiCloudHost)
+	if err != nil {
+		return workspace.TemplateRepository{}, fmt.Errorf(
+			"looking up pulumi cloud backend %s: %w\n\n%s",
+			templatePulumiCloudHost, err, howToLoginMessage)
+	}
+
+	if account.AccessToken == "" {
+		return workspace.TemplateRepository{}, fmt.Errorf(
```

Comment:
Is that ever possible? The API returns no error but an empty token?

Reply:
Yes if there are no accounts Get account won't error but will return nil. Arguably this is confusing and it should return a well known NoAccounts error. Esc had a bug a while back due to not checking for (nil,nil). Would need a careful audit if we did change that now though.